src/engine/strat_engine/thinpool/filesystem.rs

```diff
@@ -225,6 +225,6 @@ pub fn fs_usage(mount_point: &Path) -> EngineResult<(Bytes, Bytes)> {
     // Upcast to u64.
     let f_bsize = stat.block_size() as u64;
 
-    let blocks = stat.blocks();
-    Ok((Bytes(f_bsize * blocks), Bytes(f_bsize * (blocks - stat.blocks_free()))))
+    let blocks = stat.blocks() as u64;
```

Comment:
I think we've just decided to upcast everything we get from statvfs to u64, just to be safe.

What about making that very clear, as:

```
// Upcast all statvfs() result fields to u64 (on some architectures some are narrower types)
let (block_size, blocks, blocks_free) = (stat.block_size() as u64, stat.blocks() as u64, stat.blocks_free() as u64);
Ok((Bytes(block_size * blocks), Bytes(block_size * (blocks - blocks_free))))
```

Reply:
I don't think this rule would be the right way to go. I'd rather just have 32bit CI builds to catch this, see #827.

Reply:
I don't understand what you are saying.

Reply:
> I think we've just decided to upcast everything we get from statvfs to u64, just to be safe.

Maybe I misunderstood you to begin with... forget it. Let me try again.

That code suggestion is fine, I'm going to tweak the suggested comment a little, FYI.